### PR TITLE
RFC: BDD ginkgo tests 

### PR DESCRIPTION
**Pull request by MalloZup**
# RFC : add ginkgo testacc

# Description:

We might have some BDD style tests using same framework like `kubernetes` ( we could in this way upstream or merge some tests if needed)

# How to run this:

The main idea would be that one `deploy` the cluster via `deployments` dir, and then we can run the `testacc` tests for ensuring we have no regression specific to `kubic-init`

# Design/architucture:

At moment in wip, but we will have a different dir called `testacc` or any other name diff, and we use it

### CODE REVIEW

**Comment by MalloZup** [comment]
here we get like this then :  

```golang
Kubic-seeder Run basic test With more than 300 pages 
  should be a novel
  /home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:19
•
------------------------------
Kubic-seeder Run basic test Run something via ssh 
  Run on localhost ssh cmd
  /home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:25

•! Panic [2.166 seconds]
Kubic-seeder
/home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:16
  Run basic test
  /home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:17
    Run something via ssh
    /home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:24
      Run on localhost ssh cmd [It]
      /home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:25

      Test Panicked
      ssh: handshake failed: ssh: unable to authenticate, attempted methods [none password], no supported methods remain
      /usr/lib64/go/1.9/src/runtime/panic.go:491

      Full Stack Trace
      	/usr/lib64/go/1.9/src/runtime/panic.go:491 +0x283
      github.com/kubic-project/kubic-init/testacc_test.glob..func1.1.2.1()
      	/home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:31 +0x30d
      github.com/kubic-project/kubic-init/vendor/github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc420068780, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/home/dmaiocchi/go/src/github.com/kubic-project/kubic-init/testacc/seeder_test.go:13 +0xad
      testing.tRunner(0xc4201b40f0, 0x854fc0)
      	/usr/lib64/go/1.9/src/testing/testing.go:746 +0xd0
      created by testing.(*T).Run
      	/usr/lib64/go/1.9/src/testing/testing.go:789 +0x2de
      
------------------------------


Summarizing 1 Failure:

[Panic!] Kubic-seeder Run basic test Run something via ssh [It] Run on localhost ssh cmd 
/usr/lib64/go/1.9/src/runtime/panic.go:491

Ran 2 of 2 Specs in 2.167 seconds
FAIL! -- 1 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestCart (2.17s)
FAIL

Ginkgo ran 1 suite in 2.971493521s
Test Suite Failed

```

**Comment by inercia** [comment]
It looks really nice, I think BDD is the way to go...

File: Gopkg.toml [hunk 1]
```diff
@@ -118,3 +118,7 @@ required = [
 [prune]
   unused-packages = true
   go-tests = true
+
+[[constraint]]
+  name = "github.com/onsi/ginkgo"
+  version = "1.6.0"
```
Comment (inercia) [inline Gopkg.toml]: You can _port_ this constraint to the new `go.mod` file by just adding:

```
github.com/onsi/ginkgo@1.6.0
```

and running `make test` (`go` will probably rewrite this constraint and use a specific commi).

**Comment by MalloZup** [comment]
@inercia closing in favor of new one

**Pull request closed without merging**